lib/domain.js

```diff
@@ -85,6 +87,11 @@ const asyncHook = createHook({
           value: process.domain,
           writable: true
         });
+      // Because promises from other contexts don't get a domain field,
+      // the domain needs to be held alive another way. Stuffing it in a
+      // weakmap connected to the promise lifetime can fix that.
+      } else {
+        vmPromises.set(resource, process.domain);
```

Comment:
More a question than a comment 😅: should we not clean this map at one moment? (In destroy for example)

Reply:
It's a weakmap, so doesn't need cleanup like the pairings map does.

Reply:
It's a weakmap, so no cleanup is necessary. When the object used as the key is garbage-collected the link disappears allowing the value to be collected too, if there's nothing else holding it alive.

Reply:
Sorry I misread it 🙈 make sense 😅